### DISCUSSION TITLE
Order List Objects from ManyToOneRelation

### DISCUSSION THREAD

**Discussion by genfish** [pimcore/pimcore#16934] [Q&A / Support] [answered]
Hi, is it possible or what is the best practice to sort a list by a field of another class that is bound with ManyToOneRelation?

Like this example (not works):

Article has a field "user" ManyToOneRelation to User class.
User has a field "username".

```
$ArticleList = new Article\Listing();
$ArticleList->setOrderKey("user.username");
$ArticleList->setOrder("DESC");
```
Thanks!!!

**Comment by genfish** [comment, marked as answer]
Would this be a correct solution?
```
$ArticleList->onCreateQueryBuilder(function (\Doctrine\DBAL\Query\QueryBuilder $queryBuilder) use ($ArticleList) {
	$table = $ArticleList->getDao()->getTableName();
	$queryBuilder->leftJoin($table, 'object_query_$idUserTable', 'user', "user.oo_id = $table.user__id")->orderBy('user.username', 'DESC');
});
```

**Reply by markus-moser** [reply]
Yes something like this (via `onCreateQueryBuilder`) is the best possible solution to do this directly with an object list.

**Comment by genfish** [comment]
But when using `onCreateQueryBuilder`, it stops `->setCondition` from taking effect, doesn't it? Should it be added in or after?

**Reply by markus-moser** [reply]
No, `setCondition()` should still work. It should not make really a difference if you define the condition in `onCreateQueryBuilder` or with `setCondition` as the conditions are added anyways to the query builder in this method:

https://github.com/pimcore/pimcore/blob/130fd27c8a15d3cceae18fb71abf02c33db7a1b7/lib/Model/Listing/Dao/QueryBuilderHelperTrait.php#L36
